riocli/device/util.py

```diff
@@ -171,7 +177,84 @@ def is_remote_path(src, devices=[]):
     return None, src
 
 
-class DeviceNotFound(Exception):
-    def __init__(self, message='device not found'):
-        self.message = message
-        super().__init__(self.message)
+def create_hwil_device(spec, metadata):
```

Comment:
todo: type annotate this.

```suggestion
def create_hwil_device(spec: dict, metadata: dict) -> Munch:
```